misc/python/materialize/buildkite_insights/buildkite_api/generic_api.py

```diff
@@ -28,6 +35,10 @@ def get(request_path: str, params: dict[str, Any]) -> Any:
 
     url = f"{BUILDKITE_API_URL}/{request_path}"
     r = requests.get(headers=headers, url=url, params=params)
+
+    if r.status_code == STATUS_CODE_RATE_LIMIT_EXCEEDED:
```

Comment:
This is not really tested because I haven't managed to run into the limit recently...
I will do one last try with multiple concurrent runs.